### ISSUE TITLE
SA1101: "Prefix calls with this" incorrectly handles properties as expression bodies

### ISSUE THREAD

**Issue by lowkay** [DotNetAnalyzers/StyleCopAnalyzers#1501]
When using expression bodies for properties the diagnostic for SA1101 is incorrectly detected when the property name is the same as a type/enum, i.e.

```C#
    public enum Lifetime
    {
        Persistent,

        Transient,

        Scoped
    }

   public class Example
    {
        public Lifetime Lifetime => Lifetime.Persistent;
        //                          ^ Diagnostic SA1101 incorrectly detected here.
    }
```

It is not detected if this is a declared getter:

```C#

   public class Example
   {
      public Lifetime Lifetime
      {
           get
           {
               return Lifetime.Persistent;
           }
      }
   }
```

**Comment by sharwell** [comment]
*Weird* :bug: catch!

**Comment by pdelvo** [comment]
Im currently looking into this and I have trouble reproducing this. Is the code you posted exactly the one that was reporting SA1101? Which version were you using?

**Comment by lowkay** [comment]
Howdy,

It's using the latest pre-release from nuget. Take a look at the example solution here:
https://github.com/lowkay/StyleCopAnalyzerBugs/tree/master/Bug1501

**Comment by pdelvo** [comment]
![bug1501](https://cloud.githubusercontent.com/assets/1710209/9989629/79c0a7a4-605a-11e5-9f8c-13f69f7724a0.PNG)
I tried to repro this with your project but I dont see any problem. I tried both beta12 (left) and the current master (right). Maybe someone else can try to repro it?

**Comment by lowkay** [comment]
hmm, strange - no squiggle, but the build fails with SA1101 violations:

![image](https://cloud.githubusercontent.com/assets/6552669/9989780/6dd7a8b8-6053-11e5-8119-9ffdb332a9af.png)

**Comment by pdelvo** [comment]
Wow thats weird! It shows up for me in the msbuild output as well but not the error list.

**Comment by lowkay** [comment]
We've experienced some errors not getting into the error list before, always thought it was a VS bug - as we have quite a large solution and VS isn't always good at updating the error list and as I understand it adding the diagnostic is all you need to do and VS should update the relevant UX...

**Comment by pdelvo** [comment]
I am able to repro this in the debugger and I think that the bug is not our fault. Roslyns `GetSymbolInfo` in this case returns for `Lifetime` the type symbol of the enum, when run in the IDE but when run in msbuild (or our tester project) it returns the symbol of the property. I dont know why this happens. Im going to file a bug at the roslyn repo.

**Comment by lowkay** [comment]
Cool, odd that it would behave differently just because of the initiating process (MSBuild vs IDE). Will be interesting to see what the outcome is from it!

**Comment by pdelvo** [comment]
I filed the bug report. Maybe it was some last minute bug fixing in the version that shipped in VS. (or there was a bug report that introduced this bug that didnt made it into VS)

**Comment by pdelvo** [comment]
I created a PR that works around this issue for now.
